src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -365,6 +377,10 @@ private void EventLoop()
                                 // such code may keep the stack location live for longer than necessary
                                 context = null;
```

Comment:
The failure in the test `NonDisposedSocket_SafeHandlesCollected` may be because of the extra reference to the context from the wrapper, might need to also do `contextWrapper = default` to make the test work in debug builds.

Reply:
That test had been failing since Kount's PR was merged: https://github.com/dotnet/runtime/issues/35846. The JIT will often make tmp copies such that nulling out is often insufficient for debug builds. Hence my wondering why we were doing so.

Reply:
In rolling builds the seems to be failing periodically in release builds, seems like it might be something a bit different but needs to be looked into.

I'm also not too happy about nulling out the locals, seems like that's the only way the test would have passed in debug builds before my change. On one hand the test seemed fair in expecting that a non-disposed socket gets finalized. On another hand that may not happen in debuggable code at least, maybe a better approach would be to make sure it can't happen with quick JIT and enable the test only for release builds or something like that, not sure.